src/ui_ng/lib/src/repository-stackview/repository-stackview.component.ts

```diff
@@ -153,15 +168,64 @@ export class RepositoryStackviewComponent implements OnInit {
     this.clrLoad(st);
   }
 
+  saveSignatures(event: {[key: string]: string[]}): void {
+    Object.assign(this.signedCon, event);
+  }
+
   deleteRepo(repoName: string) {
+    // get children tags data
+    this.signature = '';
+    if (!(repoName in this.signedCon)) {
+      this.getTagInfo(repoName).then(() => {
+        if (this.signedNameArr.length) {
+            this.signature = this.signedNameArr.join(',');
+            this.confirmationDialogSet('DELETION_TITLE_REPO_SIGNED', repoName, 'REPOSITORY.DELETION_SUMMARY_REPO_SIGNED',   ConfirmationButtons.CLOSE);
+        }else {
+            this.confirmationDialogSet('DELETION_TITLE_REPO', repoName, 'REPOSITORY.DELETION_SUMMARY_REPO',  ConfirmationButtons.DELETE_CANCEL);
+        }
+      });
+    }else if (this.signedCon[repoName].length) {
+        this.signature = this.signedCon[repoName].join(',');
+        this.confirmationDialogSet('DELETION_TITLE_REPO_SIGNED', repoName, 'REPOSITORY.DELETION_SUMMARY_REPO_SIGNED',   ConfirmationButtons.CLOSE);
+    }else {
+        this.confirmationDialogSet('DELETION_TITLE_REPO', repoName, 'REPOSITORY.DELETION_SUMMARY_REPO',  ConfirmationButtons.DELETE_CANCEL);
+    }
+  }
+
+  getTagInfo(repoName: string): Promise<void> {
+     this.signedNameArr = [];
+     return toPromise<Tag[]>(this.tagService
+            .getTags(repoName))
+            .then(items => {
+              items.forEach((t: Tag) => {
+                if (t.signature !== null) {
+                  this.signedNameArr.push(t.name);
+                }
+              });
+              Object.assign(this.signedCon, {[repoName]: this.signedNameArr});
```

Comment:
this.signedCon[repoName]=[];

....

.then(items => {
  this.signedCon[repoName].push(t.name);
})